source/dem/dem.cc

```diff
@@ -482,11 +482,12 @@ DEMSolver<dim>::finish_simulation()
            processor_number < n_mpi_processes;
            ++processor_number)
         {
+          MPI_Barrier(MPI_COMM_WORLD);
+
           if (this_mpi_process == processor_number)
             {
               visualization_object.print_xyz(particle_handler, pcout);
             }
-          MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
Why the move of the Barrier. I don't really get it. It should work with the barrier afterwards, not before since what you want is to ensure that all processor have the time to write. Here what could happen is that the last processor does not have the time to write  output

Reply:
the problem is: with having barrier at the end, the first process does not necessarily calls write_xyz function first. then in the function, the file header (which writes "id, type, x, y, z, vx, vy, vz" before writing the particles' information) is not written in the first row. for instance the output file becomes something like:
 0 2 0.1 0.3 2 1
 1 3 0.2 0.5 2 2
id, type, x, y, z, vx, vy, vz
2 1 0.1 0.4 0.1 0.5
3 1 0.4 0.5 0.1 0.6